pkg/cluster/operation/scale_in.go

```diff
@@ -230,11 +255,8 @@ func ScaleInCluster(
 			}
 
 			if !asyncOfflineComps.Exist(instance.ComponentName()) {
-				if err := StopComponent(getter, []spec.Instance{instance}, options.OptTimeout); err != nil {
-					return errors.Annotatef(err, "failed to stop %s", component.Name())
-				}
-				if err := DestroyComponent(getter, []spec.Instance{instance}, cluster, options); err != nil {
-					return errors.Annotatef(err, "failed to destroy %s", component.Name())
+				if err := offlineInstance(component, instance); err != nil {
+					return err
 				}
 			} else {
```

Comment:
We should do the same thing for asyncOfflineComps too. Example: when you scale-in a tikv node, it will not be remove at once, instead, the PD server will transfer it's data to another tikv node, after that, the user can execute the prune command (https://github.com/pingcap/tiup/blob/master/components/cluster/command/prune.go#L45) to remove that TiKV (stop & destroy). The prune command will call `DestroyClusterTombstone` finally, so I think we should implement the same logic in DestroyClusterTombstone(https://github.com/pingcap/tiup/blob/master/pkg/cluster/operation/action.go#L178)

Reply:
implemented in daf3e3a